### PR TITLE
Alpi 66

### PR DESCRIPTION
**Pull request by JuanBarrera-Rokolabs**

### CODE REVIEW

File: Web/AlpiUS.Web/src/app/authentication/services/authentication-data.service.ts [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+import { Observable, throwError } from 'rxjs';
+import { map, catchError } from 'rxjs/operators';
+import { BaseDataService } from '../../shared/services/base-data.service';
+import { LoginResult } from '../../shared/models/login-result.model';
+import { AppConfigService } from '../../shared/services/app-config.service';
+import { AuthenticationService } from '../../shared/services/authentication.service';
+import { LoginRequest } from '../login/models/login.request';
+
+@Injectable()
+export class AuthenticationDataService extends BaseDataService {
```
Comment (dmitrii-aleksandrov) [inline Web/AlpiUS.Web/src/app/authentication/services/authentication-data.service.ts]: @JuanBarrera-Rokolabs, Why do we need 2 services? authentication.service & authentication-data.service What's difference?

File: Web/AlpiUS.WebAPI/Controllers/AuthenticationController.cs [hunk 2]
```diff
@@ -3,37 +3,44 @@
...
-        }
-
-        [HttpPost("rememberMe")]
-        public async Task<ActionResult<LoginResultModel>> RememberMe([Required] RememberMeDataModel rememberMeData)
-        {
-            var result = await this.authenticationService.RememberMe(this.mapper.Map<User.RememberMeData>(rememberMeData));
-
-            return this.mapper.Map<LoginResultModel>(result.Data);
+            return new LoginResultModel
+            {
```
Comment (dmitrii-aleksandrov) [inline Web/AlpiUS.WebAPI/Controllers/AuthenticationController.cs]: Can't we use mapper here?

File: Web/AlpiUS.WebAPI/Controllers/AuthenticationController.cs [hunk 3]
```diff
@@ -3,37 +3,44 @@
...
+        public async Task<ActionResult<LoginResultModel>> Login([Required] LoginDataModel model)
         {
-            var result = await this.authenticationService.Login(this.mapper.Map<User.LoginData>(loginData));
+            var result = await this.authenticationService.Login(this.mapper.Map<User.LoginData>(model));
 
-            return this.mapper.Map<LoginResultModel>(result.Data);
-        }
-
-        [HttpPost("rememberMe")]
-        public async Task<ActionResult<LoginResultModel>> RememberMe([Required] RememberMeDataModel rememberMeData)
```
Comment (dmitrii-aleksandrov) [inline Web/AlpiUS.WebAPI/Controllers/AuthenticationController.cs]: don't we need this method anymore?

File: Web/AlpiUS.Web/src/app/authentication/services/authentication-data.service.ts (hunk 1 above)
Comment (JuanBarrera-Rokolabs) [reply Web/AlpiUS.Web/src/app/authentication/services/authentication-data.service.ts]: For me it looked like one was the data service to call the authentication endpoints, and the other one is more like a generic user service to manage user data. I'm happy to put them back together, but it feels like two separate concerns

File: Web/AlpiUS.WebAPI/Controllers/AuthenticationController.cs (hunk 2 above)
Comment (JuanBarrera-Rokolabs) [reply Web/AlpiUS.WebAPI/Controllers/AuthenticationController.cs]: Good point, yes

File: Web/AlpiUS.WebAPI/Controllers/AuthenticationController.cs (hunk 3 above)
Comment (JuanBarrera-Rokolabs) [reply Web/AlpiUS.WebAPI/Controllers/AuthenticationController.cs]: No, I talked to Anton, and it seems that we don't need both keys, it is either one or the other (which makes sense). I'm not sure why they have two endpoints, after all the only difference seems to be in the expiry date of the session

**Pull request closed without merging**